src/app/services/user-deposit.service.ts

```diff
@@ -87,32 +85,47 @@ export class UserDepositService {
         const servicesTokenAddress$ = userDepositContract$.pipe(
             switchMap((userDepositContract) =>
                 fromPromise<string>(userDepositContract.methods.token().call())
-            ),
-            tap(() => {
-                this.servicesToken = undefined;
-            }),
-            retryWhen((errors) =>
-                errors.pipe(switchMapTo(this.raidenService.rpcConnected$))
-            ),
-            shareReplay(1)
+            )
         );
 
-        const fetchTokenInfo$ = zip(
+        const servicesTokenInfo$ = zip(
             this.raidenService.raidenAddress$,
             servicesTokenAddress$
         ).pipe(
             switchMap(([raidenAddress, tokenAddress]) =>
                 fromPromise(
                     this.tokenInfoRetriever.createBatch(
                         [tokenAddress],
-                        raidenAddress,
-                        { [tokenAddress]: this.servicesToken }
+                        raidenAddress
                     )
                 )
             ),
             map((result) => Object.values(result)[0]),
-            tap((token) => {
-                this.servicesToken = token;
+            retryWhen((errors) =>
```

Comment:
Missing knowledge from my side: will this catch the errors from the zipped `serviceTokenAddress$`? Because you removed the previous `retryWhen` above. I would guess so, just want to be sure.

Reply:
Yes it does. There is also a test for it.

Reply:
:heavy_check_mark: